_posts/2020-09-30-intro-to-hive-connector.md

```diff
@@ -0,0 +1,111 @@
+(HDFS). As cloud-based options became more prevalent, object storage that used 
+Amazon's S3 interface become more and more common and Hive expanded to include
+storing files in Amazon S3, Azure Blob Storage, Google Cloud Storage (GCS), and
+others. In order for Hive to process these files, it must have a mapping from
+SQL tables in _the runtime_ to files and directories in _the storage_ component.
+To accomplish this, Hive uses _the metastore_ to keep the metadata about the
+files such as table columns, file locations, file formats, etc... 
+
+In the early days of Big Data systems, many expected query turnaround took a 
+long time because the data that ETL operated on was big. The primary goal in
+early iterations of these systems was simply throughput over large volumes of
+data while maintaining fault-tolerance. Over time, more businesses wanted to run
+fast interactive queries over their Big Data sets instead of running jobs that
+took hours and produced possibly undesirable results. Upon Presto's creation, 
+there were 300 petabytes of data and all the metadata for this data already
+existed. The data in storage was cumbersome to move, and the data in the
+metastore would have taken a long time to repopulate in another format. The
+metastore service still functioned properly and didn't need to be changed, and 
+only the runtime that executed Hive queries needed replacement. So the creators
+took this into account and built the Presto engine to utilize the existing
+metastore metadata and files residing in storage, and the presto runtime
+effectively replaced the Hive runtime responsible for analyzing the data.
+
+![](/assets/blog/intro-to-hive-connector/presto.png)
+
+Notice that the only change in the Presto architecture is the runtime, and
+the Hive metastore still exists along with the storage. This was not by
+accident. This design was to address a common pattern that many companies faced,
+and it simplified the migration from using Hive to using Presto. Since the 
+central model used to store the metadata makes use of the Hive metastore
+regardless of the storage component used, it seemed fitting that the connector 
+received the name Hive connector since the metastore can be configured over
+different storage solutions. 
+
+Where the confusion tends to come from, is when you search for a connector
+from the context of the storage systems you want to query and may not even be 
+aware the metastore is a necessity or even exists. You will typically look for
+something like an S3 connector or GCS connector when all you really need is the
+Hive connector which connects them to a Hive metastore service, configured 
+to save metadata over the filesystem you are writing to. 
+
+To be clear, the Hive metastore service is the only Hive binary used in the
+entire Presto ecosystem when using the Hive connector. The Hive metastore is
+actually a simple service with a binary API using Thrift. This service makes
```

Comment:
Link to https://en.wikipedia.org/wiki/Apache_Thrift maybe ..

Reply:
Or https://thrift.apache.org/

also .. maybe

with a binary API using the Thrift protocol.